common/internal-types/src/protocol.rs

```diff
@@ -37,66 +34,132 @@ pub type Tag = u16;
 pub const DEFAULT_APPLICATION_TAG: Tag = 0;
 
 /// Represents packet acknowledgement
-#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[derive(Copy, Clone, Debug, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct Acknowledgement {
-    ack_signature: OffchainSignature,
-    pub ack_key_share: HalfKey,
+    #[cfg_attr(feature = "serde", serde(with = "serde_bytes"))]
+    data: [u8; Self::SIZE],
+    #[cfg_attr(feature = "serde", serde(skip))]
     validated: bool,
 }
 
+impl AsRef<[u8]> for Acknowledgement {
+    fn as_ref(&self) -> &[u8] {
+        &self.data
+    }
+}
+
+impl TryFrom<&[u8]> for Acknowledgement {
+    type Error = GeneralError;
+
+    fn try_from(value: &[u8]) -> std::result::Result<Self, Self::Error> {
+        if value.len() == Self::SIZE {
+            Ok(Self {
+                data: value.try_into().unwrap(),
+                validated: false,
+            })
+        } else {
+            Err(GeneralError::ParseError("Acknowledgement".into()))
+        }
+    }
+}
+
 impl Acknowledgement {
```

Comment:
I wonder if this should not be something like `RawAcknowledgement` and should serve as a builder through the validate function to the real acknowledgement.

Reply:
Sure, I can use a Builder pattern for it too.

Reply:
Will be done in the next PR